From d1abd1f84192453ebe451730b95a0f8416a7a86f Mon Sep 17 00:00:00 2001
From: "Philip, Avinash" <avinashphilip@ti.com>
Date: Mon, 2 Jul 2012 13:51:39 +0530
Subject: [PATCH 01/10] ARM: OMAP3+: hwmod: Corrects resource data for PWM
 devices.

PWM device has common space + module specific part in AM335x. To access
module specific part from module drivers, resource regions for PWM
device rearranged to access module specific part with index 0. This
helps module re-usability for platforms that didn't have config space
support like Davinci. Also ADDR_TYPE_RT flag removed for module specific
part.

Signed-off-by: Philip, Avinash <avinashphilip@ti.com>
---
 arch/arm/mach-omap2/omap_hwmod_33xx_data.c |   72 ++++++++++++++--------------
 1 files changed, 36 insertions(+), 36 deletions(-)

diff --git a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
index f96bbc0..b6f78c1 100644
--- a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
+++ b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
@@ -2549,17 +2549,17 @@ static struct omap_hwmod_ocp_if am33xx_l4_ls__elm = {
 
 /*
  * Splitting the resources to handle access of PWMSS config space
- * and module specific part independently
+ * and module specific part independently. Also resources being
+ * arranged to be compatible with driver
  */
 static struct omap_hwmod_addr_space am33xx_ehrpwm0_addr_space[] = {
 	{
-		.pa_start	= 0x48300000,
-		.pa_end		= 0x48300000 + SZ_16 - 1,
-		.flags		= ADDR_TYPE_RT
-	},
-	{
 		.pa_start	= 0x48300200,
 		.pa_end		= 0x48300200 + SZ_256 - 1,
+	},
+	{
+		.pa_start	= 0x48300000,
+		.pa_end		= 0x48300000 + SZ_16 - 1,
 		.flags		= ADDR_TYPE_RT
 	},
 	{ }
@@ -2575,17 +2575,17 @@ static struct omap_hwmod_ocp_if am33xx_l4_ls__ehrpwm0 = {
 
 /*
  * Splitting the resources to handle access of PWMSS config space
- * and module specific part independently
+ * and module specific part independently. Also resources being
+ * arranged to be compatible with driver
  */
 static struct omap_hwmod_addr_space am33xx_ehrpwm1_addr_space[] = {
 	{
-		.pa_start	= 0x48302000,
-		.pa_end		= 0x48302000 + SZ_16 - 1,
-		.flags		= ADDR_TYPE_RT
-	},
-	{
 		.pa_start	= 0x48302200,
 		.pa_end		= 0x48302200 + SZ_256 - 1,
+	},
+	{
+		.pa_start	= 0x48302000,
+		.pa_end		= 0x48302000 + SZ_16 - 1,
 		.flags		= ADDR_TYPE_RT
 	},
 	{ }
@@ -2601,17 +2601,17 @@ static struct omap_hwmod_ocp_if am33xx_l4_ls__ehrpwm1 = {
 
 /*
  * Splitting the resources to handle access of PWMSS config space
- * and module specific part independently
+ * and module specific part independently. Also resources being
+ * arranged to be compatible with driver
  */
 static struct omap_hwmod_addr_space am33xx_ehrpwm2_addr_space[] = {
 	{
-		.pa_start	= 0x48304000,
-		.pa_end		= 0x48304000 + SZ_16 - 1,
-		.flags		= ADDR_TYPE_RT
-	},
-	{
 		.pa_start	= 0x48304200,
 		.pa_end		= 0x48304200 + SZ_256 - 1,
+	},
+	{
+		.pa_start	= 0x48304000,
+		.pa_end		= 0x48304000 + SZ_16 - 1,
 		.flags		= ADDR_TYPE_RT
 	},
 	{ }
@@ -2627,17 +2627,17 @@ static struct omap_hwmod_ocp_if am33xx_l4_ls__ehrpwm2 = {
 
 /*
  * Splitting the resources to handle access of PWMSS config space
- * and module specific part independently
+ * and module specific part independently. Also resources being
+ * arranged to be compatible with driver
  */
 static struct omap_hwmod_addr_space am33xx_ecap0_addr_space[] = {
 	{
-		.pa_start	= 0x48300000,
-		.pa_end		= 0x48300000 + SZ_16 - 1,
-		.flags		= ADDR_TYPE_RT
-	},
-	{
 		.pa_start	= 0x48300100,
 		.pa_end		= 0x48300100 + SZ_256 - 1,
+	},
+	{
+		.pa_start	= 0x48300000,
+		.pa_end		= 0x48300000 + SZ_16 - 1,
 		.flags		= ADDR_TYPE_RT
 	},
 	{ }
@@ -2653,17 +2653,17 @@ static struct omap_hwmod_ocp_if am33xx_l4_ls__ecap0 = {
 
 /*
  * Splitting the resources to handle access of PWMSS config space
- * and module specific part independently
+ * and module specific part independently. Also resources being
+ * arranged to be compatible with driver
  */
 static struct omap_hwmod_addr_space am33xx_ecap1_addr_space[] = {
 	{
-		.pa_start	= 0x48302000,
-		.pa_end		= 0x48302000 + SZ_16 - 1,
-		.flags		= ADDR_TYPE_RT
-	},
-	{
 		.pa_start	= 0x48302100,
 		.pa_end		= 0x48302100 + SZ_256 - 1,
+	},
+	{
+		.pa_start	= 0x48302000,
+		.pa_end		= 0x48302000 + SZ_16 - 1,
 		.flags		= ADDR_TYPE_RT
 	},
 	{ }
@@ -2679,17 +2679,17 @@ static struct omap_hwmod_ocp_if am33xx_l4_ls__ecap1 = {
 
 /*
  * Splitting the resources to handle access of PWMSS config space
- * and module specific part independently
+ * and module specific part independently. Also resources being
+ * arranged to be compatible with driver
  */
 static struct omap_hwmod_addr_space am33xx_ecap2_addr_space[] = {
 	{
-		.pa_start	= 0x48304000,
-		.pa_end		= 0x48304000 + SZ_16 - 1,
-		.flags		= ADDR_TYPE_RT
-	},
-	{
 		.pa_start	= 0x48304100,
 		.pa_end		= 0x48304100 + SZ_256 - 1,
+	},
+	{
+		.pa_start	= 0x48304000,
+		.pa_end		= 0x48304000 + SZ_16 - 1,
 		.flags		= ADDR_TYPE_RT
 	},
 	{ }
-- 
1.7.7.6

